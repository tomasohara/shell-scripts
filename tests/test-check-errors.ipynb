{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a8eacedb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## Bracketed Paste is disabled to prevent characters after output\n",
    "## Example: \n",
    "## $ echo 'Hi'\n",
    "## | Hi?2004l\n",
    "\n",
    "# bind 'set enable-bracketed-paste off'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ab1de4e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PROMPT_NSGMXBMINMVZ\\[\\]>\n"
     ]
    }
   ],
   "source": [
    "## TEST:  Make sure simple prompt used (e.g., no escapes that might contaminate output)\n",
    "## PS1=\"$ \"\n",
    "## TODO: PS1=\"> \"\n",
    "\n",
    "## NOTE: The Jupyter bash kernel requires that PS1 not be modified as they customize it. \n",
    "echo $PS1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f2fdf1c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD (SUPER BAD)\n",
    "# # Delete all aliases and function\n",
    "# # TODO: Instead start from pristine environment\n",
    "# unalias -a\n",
    "# for f in $(typeset -f | egrep '^\\w+'); do unset -f $f; done\n",
    "# BIN_DIR=$PWD/..\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d1c92493",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting a temp directory for tests\n",
    "TMP=/tmp/test-check-errors\n",
    "\n",
    "## NOTE: Source it directly from the ./tests directory.\n",
    "BIN_DIR=$PWD/.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "da176b40",
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD\n",
    "# temp_dir=$TMP/test-2334\n",
    "# mkdir -p \"$temp_dir\"\n",
    "# # TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "# cd \"$temp_dir\"\n",
    "# pwd\n",
    "\n",
    "temp_dir=$TMP/test-2334\n",
    "trash_dir=$TMP/\"_temp-trash-$(date \"+%Y%m%d%H%M%S\")\"\n",
    "\n",
    "command mkdir -p \"$temp_dir\"\n",
    "command mkdir -p \"$trash_dir\"\n",
    "# TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "command cd \"$temp_dir\"\n",
    "\n",
    "## OLD: Debug\n",
    "# command pwd\n",
    "\n",
    "## OLD\n",
    "## ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)\n",
    "# alias linebr=\"printf '%*s\\n' \"${COLUMNS:-$(tput cols)}\" '' | tr ' ' -\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "eb29a5b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "# alias testnum=\"sed -r \"s/[0-9]/X/g\"\" \n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/user/g\"\"\n",
    "## For number of digits not being critical\n",
    "alias testnum=\"sed -r \"s/[0-9][0-9]*/N/g\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9de59cca",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/tmp/test-check-errors/test-2334\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "# command pwd | testuser\n",
    "command pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9abcb30c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "## TODO: Find a way to remove the full path \n",
    "# python $BIN_DIR/check_errors.py | grep -q \"ruby\" && echo \"True\" || echo \"False\" > /dev/null \n",
    "python $BIN_DIR/check_errors.py 2>&1 | grep -q \"ruby\" && echo \"True\" || echo \"False\" > /dev/null "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f965b950",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: mezcla in /home/aveey/.local/lib/python3.10/site-packages (1.3.9.6)\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "## pip3 install mezcla | testnum > /dev/null\n",
    "## Explicit approach (debian-like): pip3 install mezcla --break-system-packages > /dev/null\n",
    "pip3 install mezcla 2> /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1f464ae8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "## NEW\n",
    "pip3 show mezcla | grep -q \"tomasohara@gmail.com\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "91d951af",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# check_errors.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5705f9dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# python3 $BIN_DIR/check_errors.py ./test_check_errors.py\n",
    "# # Error: TypeError: 'NoneType' object is not iterable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3ff3937e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# check_errors.py ./test_check_errors.py --verbose\n",
    "# # Error: TypeError: 'NoneType' object is not iterable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "46dcd2c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "## TODO: testuser doesn't work below\n",
    "## TODO: Find a way to remove the full path \n",
    "\n",
    "## OLD\n",
    "# python3 $BIN_DIR/check_errors.py -h | testuser\n",
    "\n",
    "python3 $BIN_DIR/check_errors.py -h 2>&1 | grep -q \"ruby\" && echo \"True\" || echo \"False\"\n",
    "## TODO: Remove the complete path (along with user) of the check_errors.py script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e73248ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD\n",
    "# check_errors.py ./test_extract_matches.py --verbose | testuser\n",
    "## Error: TypeError: 'NoneType' object is not iterable\n",
    "\n",
    "## TODO: \n",
    "\n",
    "# command python3 $BIN_DIR/check_errors.py $BIN_DIR/tests/test_extract_matches.py --verbose | testuser\n",
    "\n",
    "python3 $BIN_DIR/check_errors.py $BIN_DIR/tests/test_extract_matches.py --verbose > test_extract_matches_verbose.out 2>&1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "51a64432",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "# check_errors.py --warning ./test_extract_matches.py | testuser | testnum\n",
    "# # Error: TypeError: 'NoneType' object is not iterable\n",
    "\n",
    "cat test_extract_matches_verbose.out | grep -q \"Deleting any temporary files\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "022992db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "cat test_extract_matches_verbose.out | grep -q \"himom\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "16dbbf36",
   "metadata": {},
   "outputs": [],
   "source": [
    "# EXPECTED OUTPUT FOR ABOVE COMMAND \n",
    "\n",
    "# Processing input\n",
    "# 26           def test_python_error(self):\n",
    "# 27               \"\"\"check python error\"\"\"\n",
    "# 28               input_string    = 'python -c \"print(1\\\\2)\" 2>&1'\n",
    "# 29   >>>         expected_result = '1          File \"<string>\", line 1\\n2            print(1\\\\2)\\n3                     ^\\n4    >>> SyntaxError: unexpected character after line continuation character <<<' <<<\n",
    "# 30               self.assertEqual(gh.run(f'{input_string} | {SCRIPT} -'), expected_result)\n",
    "# 31       \n",
    "# 32       \n",
    "\n",
    "# 48           def test_context_lines(self):\n",
    "# 49               \"\"\"test context lines\"\"\"\n",
    "# 50               input_string     = 'python -c \"print(1\\\\2)\" 2>&1'\n",
    "# 51   >>>         result_context_1 = '3                     ^\\n4    >>> SyntaxError: unexpected character after line continuation character <<<' <<<\n",
    "# 52   >>>         result_context_2 = '2            print(1\\\\2)\\n3                     ^\\n4    >>> SyntaxError: unexpected character after line continuation character <<<' <<<\n",
    "# 53               self.assertEqual(gh.run(f'{input_string} | {SCRIPT} --context 1 -'), result_context_1)\n",
    "# 54               self.assertEqual(gh.run(f'{input_string} | {SCRIPT} --context 2 -'), result_context_2)\n",
    "# 55"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "664a28ac",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# # WORKS FOR test_check_errors.py\n",
    "\n",
    "# # WORKS FINE (ERROR PRONE ON BATSPP)\n",
    "# # $ python3 $BIN_DIR/check_errors.py --warning ./test_check_errors.py\n",
    "# check_errors.py --warning ./test_check_errors.py | testuser | testnum\n",
    "\n",
    "# # Error: TypeError: 'NoneType' object is not iterable\n",
    "## OLD\n",
    "# python $BIN_DIR/check_errors.py --warning $BIN_DIR/tests/test_check_errors.py > check_errors.output\n",
    "python $BIN_DIR/check_errors.py --warning $BIN_DIR/tests/test_check_errors.py > check_errors.output 2>&1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b200bada",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "cat check_errors.output | grep -q \"test_python_error\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f8a387ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# BAD: /bin/rm -rf $temp_dir > /dev/null\n",
    "command mv -f * \"$trash_dir\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
