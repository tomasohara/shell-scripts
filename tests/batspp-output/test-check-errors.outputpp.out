1..10
not ok 1 test-1
# (in test file batspp-output/test-check-errors.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== echo $PS1
#  ==========
#
# ========= [PEXP\[\]ECT_PROMPT> =========
# [PEXP\[\]ECT_PROMPT>
# ============================
not ok 2 test-3
# (in test file batspp-output/test-check-errors.outputpp, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== unalias -a
#  ==========
# ========= $ for f in ; do unset -f ; done\n$ BIN_DIR=/tmp/batspp-77841/test-3/.. =========
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-10-actual ()
# test-10-expected ()
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
# END_EXPECTED
# test-11-actual ()
# test-11-expected ()
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# test_test-2d11 ()
# test_test-2d3 ()
# test_test-2d4 ()
# test_test-2d5 ()
# test_test-2d6 ()
# test_test-2d7 ()
# test_test-2d8 ()
# test_test-2d9 () ; do unset -f ; done
# $ BIN_DIR=/tmp/batspp-77841/test-3/..
# ============================
not ok 3 test-4
# (in test file batspp-output/test-check-errors.outputpp, line 94)
#   `[ "$(test-4-actual)" == "$(test-4-expected)" ]' failed
# ========== TMP=/tmp/test-check-errors
#  ==========
# ========= ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=$PWD/.. =========
# ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=/tmp/batspp-77841/test-4/..
# ============================
not ok 4 test-5
# (in test file batspp-output/test-check-errors.outputpp, line 129)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== temp_dir=$TMP/test-2334
#  ==========
# ========= $ mkdir -p "$temp_dir"
# # TODO: /bin/rm -rvf "$temp_dir"
# $ cd "$temp_dir"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -"
# /tmp/test-check-errors/test-2334 =========
# $ mkdir -p "/test-2334"
# # TODO: /bin/rm -rvf "/test-2334"
# $ cd "/test-2334"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "80" '' | tr ' ' -"
# /tmp/test-check-errors/test-2334
# ============================
not ok 5 test-6
# (in test file batspp-output/test-check-errors.outputpp, line 169)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== alias testnum="sed -r "s/[0-9]/X/g""
#  ==========
# ========= $ alias testuser="sed -r "s/"$USER"+/user/g"" =========
# $ alias testuser="sed -r "s/"aveey"+/user/g""
# ============================
not ok 6 test-7
# (from function `test-7-actual' in file batspp-output/test-check-errors.outputpp, line 210,
#  in test file batspp-output/test-check-errors.outputpp, line 197)
#   `test-7-actual ' failed with status 127
# ========== pwd | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-check-errors.outputpp: line 210: testuser: command not found
not ok 7 test-8
# (in test file batspp-output/test-check-errors.outputpp, line 237)
#   `[ "$(test-8-actual)" == "$(test-8-expected)" ]' failed
# ========== check_errors.py
#  ==========
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
#                        [--no_asterisks] [--ruby] [--skip_ruby_lib] [--relaxed]
#                        [--strict] [--verbose] [--context CONTEXT]
#                        [filename]
#
# Scan the error log for errors, warnings and other suspicious results.
# This prints the offending line bracketted by >>> and <<< along with N
# lines before and after to provide context.
#
# ex: check_errors.py whatever
#
# Notes:
#
# - The default context is 1
#
# - Warnings are skipped by default
#
# - Use -no_asterisks if input uses ***'s outside of error contexts
#
# Use -relaxed to exclude special cases (e.g., xyz='error')
#
# positional arguments:
#   filename           Input filename
#
# options:
#   -h, --help         show this help message and exit
#   --warning          alias for -warnings
#   --warnings         include warnings?
#   --skip_warnings    omit warnings?
#   --no_asterisks     skip warnings for "***" in text
#   --ruby             alias for -skip_ruby_lib
#   --skip_ruby_lib    skip Ruby library related errors
#   --relaxed          relaxed for special cases
#   --strict           alias for relaxed=0
#   --verbose          show more details
#   --context CONTEXT  context lines before and after
#
# Notes:
# - Use - for filename to skip usage (i.e., a la stdin).
# - Use "ENV1='v1' ENV2='v2' python .../check_errors.py ..." for environment options.
# ========= usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
#                        [--no_asterisks] [--ruby] [--skip_ruby_lib] [--relaxed]
#                        [--strict] [--verbose] [--context CONTEXT]
#                        [filename] =========
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
#                        [--no_asterisks] [--ruby] [--skip_ruby_lib] [--relaxed]
#                        [--strict] [--verbose] [--context CONTEXT]
#                        [filename]
# ============================
not ok 8 test-9
# (from function `test-9-actual' in file batspp-output/test-check-errors.outputpp, line 281,
#  in test file batspp-output/test-check-errors.outputpp, line 268)
#   `test-9-actual ' failed with status 127
# ========== pip3 install mezcla | testnum > /dev/null
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-check-errors.outputpp: line 281: testnum: command not found
# ERROR: Pipe to stdout was broken
# Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
# BrokenPipeError: [Errno 32] Broken pipe
not ok 9 test-10
# (in test file batspp-output/test-check-errors.outputpp, line 308)
#   `[ "$(test-10-actual)" == "$(test-10-expected)" ]' failed
# ========== check_errors.py
#  ==========
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
#                        [--no_asterisks] [--ruby] [--skip_ruby_lib] [--relaxed]
#                        [--strict] [--verbose] [--context CONTEXT]
#                        [filename]
#
# Scan the error log for errors, warnings and other suspicious results.
# This prints the offending line bracketted by >>> and <<< along with N
# lines before and after to provide context.
#
# ex: check_errors.py whatever
#
# Notes:
#
# - The default context is 1
#
# - Warnings are skipped by default
#
# - Use -no_asterisks if input uses ***'s outside of error contexts
#
# Use -relaxed to exclude special cases (e.g., xyz='error')
#
# positional arguments:
#   filename           Input filename
#
# options:
#   -h, --help         show this help message and exit
#   --warning          alias for -warnings
#   --warnings         include warnings?
#   --skip_warnings    omit warnings?
#   --no_asterisks     skip warnings for "***" in text
#   --ruby             alias for -skip_ruby_lib
#   --skip_ruby_lib    skip Ruby library related errors
#   --relaxed          relaxed for special cases
#   --strict           alias for relaxed=0
#   --verbose          show more details
#   --context CONTEXT  context lines before and after
#
# Notes:
# - Use - for filename to skip usage (i.e., a la stdin).
# - Use "ENV1='v1' ENV2='v2' python .../check_errors.py ..." for environment options.
# ========= usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
#                        [--no_asterisks] [--ruby] [--skip_ruby_lib] [--relaxed]
#                        [--strict] [--verbose] [--context CONTEXT]
#                        [filename] =========
# usage: check_errors.py [-h] [--warning] [--warnings] [--skip_warnings]
#                        [--no_asterisks] [--ruby] [--skip_ruby_lib] [--relaxed]
#                        [--strict] [--verbose] [--context CONTEXT]
#                        [filename]
# ============================
not ok 10 test-11
# (from function `test-11-actual' in file batspp-output/test-check-errors.outputpp, line 352,
#  in test file batspp-output/test-check-errors.outputpp, line 339)
#   `test-11-actual ' failed with status 2
# ========== python3 $BIN_DIR/check_errors.py -h
#  ==========
# python3: can't open file '/check_errors.py': [Errno 2] No such file or directory
